Modules/CetTest.cmake

```diff
@@ -1009,6 +1038,41 @@ function(cet_test_env)
   set(CET_TEST_ENV "${CET_TEST_ENV}" PARENT_SCOPE)
 endfunction()
 
+#[================================================================[.rst:
+.. command:: cet_test_prepend_env
+
+   Prepend to path-like environment variables to the test environment
+   for tests defined in and below the current directory.
+
+   .. code-block:: cmake
+
+      cet_test_prepend_env(<var> [REMOVE_DUPLICATES] <dir> ...)
+
+   Options
+   ^^^^^^^
+
+   ``REMOVE_DUPLICATES``
+     Remove any duplicate directory entries that may appear in the
+     specified directories.  For a given duplicate, only the first
+     entry will be kept.
+
+#]================================================================]
+
+function(cet_test_prepend_env CET_ENV_VAR)
+  cmake_parse_arguments(PARSE_ARGV 1 CET_TEST "REMOVE_DUPLICATES" "" "")
+  if (CET_TEST_REMOVE_DUPLICATES)
+    list(REMOVE_DUPLICATES CET_TEST_UNPARSED_ARGUMENTS)
+  endif()
+  string(REPLACE ";" ":" PREFIX "${CET_TEST_UNPARSED_ARGUMENTS}")
+  if (DEFINED ENV{${CET_ENV_VAR}})
+    set(PATHS "${CET_ENV_VAR}=${PREFIX}:$ENV{${CET_ENV_VAR}}")
+  else()
+    set(PATHS "${CET_ENV_VAR}=${PREFIX}")
+  endif()
+  list(APPEND CET_TEST_ENV "${PATHS}")
+  set(CET_TEST_ENV "${CET_TEST_ENV}" PARENT_SCOPE)
```

Comment:
I worry that this implementation does not play well with `cet_test_env()`: if there is already an entry for this environment variable in `CET_TEST_ENV` we should prepend to that value rather than `$ENV{${CET_ENV_VAR}}`

Reply:
Okay, that's fair.  That just means an extra check of what's in `CET_TEST_ENV`...but those are`"<var>=<value>"` entries, which will need to be parsed, right?

Reply:
Yeah. I'm having a bit of a play rn, because it also occurs to me that `cet_test_env()` is prone to duplication, which would be a pain to have to handle in `cet_test_prepend_env()`